sysconfig/cloudinit.go

```diff
@@ -112,12 +112,18 @@ func configureCloudInit(opts *Options) (err error) {
 
 	// next check if there is a gadget cloud.conf to install
 	if HasGadgetCloudConf(opts.GadgetDir) {
-		// then copy / install the gadget config and return without considering
-		// CloudInitSrcDir
-		// TODO:UC20: we may eventually want to consider both CloudInitSrcDir
-		// and the gadget cloud.conf so returning here may be wrong
+		// then copy / install the gadget config first
 		gadgetCloudConf := filepath.Join(opts.GadgetDir, "cloud.conf")
-		return installGadgetCloudInitCfg(gadgetCloudConf, WritableDefaultsDir(opts.TargetRootDir))
+		err := installGadgetCloudInitCfg(gadgetCloudConf, WritableDefaultsDir(opts.TargetRootDir))
+		if err != nil {
+			return err
+		}
+
+		// we don't return here to enable also copying any cloud-init config
+		// from ubuntu-seed in order for both to be used simultaneously for
+		// example on test devices where the gadget has a gadget.yaml, but for
+		// testing purposes you also want to provision another user with
+		// ubuntu-seed cloud-init config
```

Comment:
right now the gadget will use 80_device_gadget.cfg so likely will have higher precedence than the ubuntu-seed bits, does that make sense? shouldn't it be the reverse?  either way we probably need to rename things to make sure that the seed bits are all before or all after

Reply:
I thought about this a bit, but I can argue both ways for which one should be higher priority. At least without renaming files, the deployment has a choice in that folks can rename the ubuntu-seed one to be higher or lower in priority than the gadget for their use case.

Reply:
I asked on the bug for the reporter's opinion on what they need. We can go with what they need for now and accommodate other situations as folks request them, since this is the first and only such request for both gadget + ubuntu-seed configuration.

Reply:
The folks on the bug say that they would like to have ubuntu-seed take precedence, so I will adjust this accordingly